eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -104,11 +105,51 @@ object InteractiveTxBuilder {
 
   case class RequireConfirmedInputs(forLocal: Boolean, forRemote: Boolean)
 
+  /** An input that is already shared between participants (e.g. the current funding output when doing a splice). */
+  sealed trait SharedInput {
+    // @formatter:off
+    def info: InputInfo
+    def localAmount: Satoshi
+    def remoteAmount: Satoshi
+    def weight: Long
+    // @formatter:on
+  }
+
+  case class Multisig2of2Input(info: InputInfo, localAmount: Satoshi, remoteAmount: Satoshi, localFundingPubkey: PublicKey, remoteFundingPubkey: PublicKey) extends SharedInput {
+    // This value was computed assuming 73 bytes signatures (worst-case scenario).
+    override val weight: Long = 388
+  }
+
+  object Multisig2of2Input {
+    def apply(keyManager: ChannelKeyManager, params: Params, commitment: Commitment): Multisig2of2Input = Multisig2of2Input(
+      info = commitment.commitInput,
+      localAmount = commitment.localCommit.spec.toLocal.truncateToSatoshi,
+      remoteAmount = commitment.localCommit.spec.toRemote.truncateToSatoshi,
+      localFundingPubkey = keyManager.fundingPublicKey(params.localParams.fundingKeyPath).publicKey,
+      remoteFundingPubkey = params.remoteParams.fundingPubKey
+    )
+  }
+
+  /**
+   * @param channelId              id of the channel.
+   * @param isInitiator            true if we initiated the protocol, in which case we will pay fees for the shared parts of the transaction.
+   * @param localAmount            amount contributed by us to the shared output.
+   * @param remoteAmount           amount contributed by our peer to the shared output.
+   * @param sharedInput_opt        previous input shared between the two participants (e.g. previous funding output when splicing).
+   * @param fundingPubkeyScript    script of the shared output.
+   * @param localOutputs           outputs to be added to the shared transaction (e.g. splice-out).
+   * @param lockTime               transaction lock time.
+   * @param dustLimit              outputs below this value are considered invalid.
+   * @param targetFeerate          transaction feerate.
+   * @param requireConfirmedInputs we may require that inputs added to the transaction are confirmed, especially with peers we don't trust.
+   */
   case class InteractiveTxParams(channelId: ByteVector32,
                                  isInitiator: Boolean,
                                  localAmount: Satoshi,
                                  remoteAmount: Satoshi,
+                                 sharedInput_opt: Option[SharedInput],
```

Comment:
Investigate if we can move `sharedInput_opt` and `localOutputs` to the `purpose` field?